gitensite/apps/bookinfo/db.py

```diff
@@ -0,0 +1,41 @@
+from gitensite.apps.bookinfo.models import Author, Book
+
+import yaml as PyYAML
+def default_ctor(loader, tag_suffix, node):
+    return tag_suffix + ' ' + node.value
+PyYAML.add_multi_constructor('!lcc', default_ctor)
+PyYAML.add_multi_constructor('!lcsh', default_ctor)
+
+def addBookFromYaml(yaml):
+    obj = PyYAML.load(yaml)
+
+    (book,created) = Book.objects.get_or_create(book_id=int(obj['identifiers']['gutenberg']), repo_name=obj['_repo'])
```

Comment:
book_id is declared unique, so this will throw exceptions if you have two repos with same book_id and different repo_name. Unfortunately these exist; also a problem if repo_names change

Reply:
Ok, thanks for the warning. I guess we will have to find a different approach for the id. Maybe we will just use a numeric ID that increments each time a book is added.

Reply:
I don't think you need a different approach, I think all you need to do is drop the unique declaration.